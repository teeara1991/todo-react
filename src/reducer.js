export function reducer(state, action) {
  let todos = [...state.todos];
  const index = action.payload.index;
  switch (action.type) {
    case "add todo":
      const name = action.payload.data;
      todos.push({ name, checked: false, completed: false });
      return { ...state, todos };
    case "delete todo":
      todos.splice(index, 1);
      return { ...state, todos };
    case "complete todo":
      todos[index].completed = !todos[index].completed;
      todos[index].completed
        ? (todos[index].checked = true)
        : (todos[index].checked = false);
      return { ...state, todos };
    case "load todos":
        const data = action.payload.data;
        todos = (data && [...data]) || [];
        return { ...state, todos }; 
    default:
      throw new Error();
  }
}

export const initialState = {
  todos: []
};


