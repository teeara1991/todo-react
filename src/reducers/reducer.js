import {
  ADD_TODO,
  DELETE_TODO,
  COMPLETE_TODO,
  LOAD_TODOS
} from '../constants/constants';

export function reducer(state = [], action) {
  let todos = [...state.todos];
  switch (action.type) {
    case ADD_TODO:
      const name = action.payload.data;
      todos.push({ name, checked: false, completed: false });
      return { ...state, todos };
    case DELETE_TODO:
      todos.splice(action.payload.index, 1);
      return { ...state, todos };
    case COMPLETE_TODO:
      const { index } = action.payload;
      todos[index].completed = !todos[index].completed;
      todos[index].completed
        ? (todos[index].checked = true)
        : (todos[index].checked = false);
      return { ...state, todos };
    case LOAD_TODOS:
      const { data } = action.payload;
      todos = (data && [...data]) || [];
      return { ...state, todos };
    default:
      return state;
  }
}

export const initialState = {
  todos: []
};
