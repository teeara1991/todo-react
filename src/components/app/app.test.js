import React from "react";
import App from "./app";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { initialState, reducer } from "../../reducers/reducer";
import toJson from 'enzyme-to-json'
import {
    addTodo,
    deleteTodo,
    completeTodo,
} from "../../actions/actions";

configure({ adapter: new Adapter() });

const store = createStore(reducer, initialState);

it('calls addTodo Redux action creator return new state', () => {
    const firstState = {
      todos: [{name: 'Buy groceries',completed: false,checked: false}]
    }
    const lastState = {
        todos: [{name: 'Buy groceries',completed: false,checked: false},
         {name: 'Change oil', completed:false,checked: false}]
    }
    expect(reducer(firstState, addTodo('Change oil'))).toStrictEqual(lastState);
})

it('calls deleteTodo Redux action creator return new state', () => {
    const firstState = {
        todos: [{name: 'Buy groceries',completed: false,checked: false},
        {name: 'Change oil', completed:false,checked: false}]
    }
    const lastState = {
        todos: [{name: 'Buy groceries',completed: false,checked: false}]
    }
    expect(reducer(firstState, deleteTodo(1))).toStrictEqual(lastState);
})

it('calls completeTodo Redux action creator return new state', () => {
    const firstState = {
        todos: [{name: 'Buy groceries', completed: false,checked: false},
        {name: 'Change oil', completed:false,checked: false}]
    }
    const lastState = {
        todos: [{name: 'Buy groceries', completed: true, checked: true},
        {name: 'Change oil', completed:false,checked: false}]
    }
    expect(reducer(firstState, completeTodo(0))).toStrictEqual(lastState);
})

it('matches the snapshot', () => {
  const tree = mount(<Provider store={store}>
    <App />
  </Provider>)
  expect(toJson(tree)).toMatchSnapshot()
})


 
