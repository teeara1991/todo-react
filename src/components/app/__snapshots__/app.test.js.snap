// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(App)>
    <App
      onComplete={[Function]}
      onDelete={[Function]}
      onLoad={[Function]}
      onSubmit={[Function]}
      todos={Array []}
    >
      <div>
        <TodoInput
          onSubmit={[Function]}
        >
          <form
            name="todo"
            onSubmit={[Function]}
          >
            <div
              className="input-group mb-3"
            >
              <input
                aria-describedby="basic-addon2"
                autoFocus={true}
                className="form-control"
                name="todo_name"
                placeholder="Добавить задачу"
                type="text"
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-primary btn-outline-secondary"
                  type="submit"
                >
                  +
                </button>
              </div>
            </div>
          </form>
        </TodoInput>
        <TodoList
          onComplete={[Function]}
          onDelete={[Function]}
          todos={Array []}
        >
          <ul
            className="todos todos-custom"
          />
        </TodoList>
      </div>
    </App>
  </Connect(App)>
</Provider>
`;
